More Random Edition/NPCPreferences/PreferenceRandomizer.cs

```diff
@@ -0,0 +1,317 @@
+                string name = NPC;
+
+                for (int index = 1; index <= 9; index += 2) { tokens[index] = GetPreferenceString(index, unusedCategories, unusedItems); }
+
+                replacements.Add(name, string.Join("/", tokens));
+            }
+
+            // Update Loves/Hates for Bundle reqs
+            UpdateBundlePrefs(replacements);
+
+            WriteToSpoilerLog(replacements);
+            return replacements;
+        }
+
+        /// <summary>
+        /// Builds universal preference string
+        /// </summary>
+        /// <param name="unusedCategories">Keeps track of which categories have not yet been assigned</param>
+        /// <param name="unusedItems">Keeps track of which items have not yet been assigned.</param>
+        /// <returns><c>string</c> containing IDs of categories and items</returns>
+        private static string GetUniversalPreferenceString(List<int> unusedCategories, List<Item> unusedItems)
+        {
+            // No need to vary quantities per index.May end up with lots of loved items, lots of hated items, both, neither, etc.
+            int catNum = Range.GetRandomValue(0, 10);
+            int itemNum = Range.GetRandomValue(5, 30);
+
+            string catString = "";
+            string itemString = "";
+
+            while (unusedCategories.Any() && catNum > 0)
+            {
+                catString += Globals.RNGGetAndRemoveRandomValueFromList(unusedCategories) + " ";
+                catNum--;
+            }
+
+            while (unusedItems.Any() && itemNum > 0)
+            {
+                itemString += Globals.RNGGetAndRemoveRandomValueFromList(unusedItems).Id + " ";
+                itemNum--;
+            }
+
+            return (catString + itemString).Trim();
+        }
+
+        /// <summary>
+        /// Builds an NPC's preference string for a given index (loves, hates, etc.).
+        /// </summary>
+        /// <param name="index">the index of the NPC's prefstring.</param>
+        /// <param name="unusedCategories">Holds list of categories which have not yet been assigned - prevents double-assignment.</param>
+        /// <param name="unusedItems">Holds list of Items which have not yet been assigned - prevents double-assignment.</param>
+        /// <returns>NPC's preference string for a given index.</returns>
+        private static string GetPreferenceString(int index, List<int> unusedCategories, List<Item> unusedItems)
+        {
+            int minItems;
+            int maxItems;
+
+            // Should probably be moved into its own function
+            // Determine how many items to add per category - data available here: https://pastebin.com/gFEduBVd
+            // Basically, add more loved/liked items than hated/disliked, and few neutrals
+            switch (index)
+            {
+                // Loved Items - higher minimum, so generated bundles have more items to draw from
+                case 1:
```

Comment:
Can you change these case numbers to be constants or something so it's easier to tell what they're for?

Reply:
Yep, taken care of - those were placeholders that never got un-placeholdered 😅